### PR TITLE
Feature 64

### LINKED ISSUE #64: As a guardian I would like to update the settings for a given citizen so that it fits their preference in terms in colors

### PR DESCRIPTION
**Pull request by Simkortet**
Fixes #64

It is now possible to select a color theme to display in the weekplan screen. Three standard color themes have been added as described in the Issue.

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw607f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [x] All new functionality is covered by tests
- [x] It is understandable, by reading the tests, what they do
- [x] The tests covers sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [x] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw609f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by fredbabe — APPROVED** [review]

File: lib/screens/weekplan_screen.dart [hunk 1]
```diff
@@ -285,17 +285,20 @@ class WeekplanScreen extends StatelessWidget {
                     final int _daysToDisplay = _settingsModel.nrOfDaysToDisplay;
 
                     // If the option of showing 1 day is chosen the
-                    // _weekdayCounter
-                    // must start from today's date
+                    // _weekdayCounter must start from today's date
```
Comment (joandrsn) [inline lib/screens/weekplan_screen.dart:288]: Why is this line changed?

File: lib/screens/settings_screens/settings_screen.dart [hunk 2]
```diff
@@ -25,19 +36,42 @@ class SettingsScreen extends StatelessWidget {
...
+                    ).then((Object object) =>
+                          _settingsBloc.loadSettings(_user)),
+                titleTrailing: ThemeBox.fromHexValues(
+                    settingsModel.weekDayColors[0].hexColor,
+                    settingsModel.weekDayColors[1].hexColor
+                )
+            ),
+            SettingsArrowButton('Tegn for udførelse', () {})
+          ]);
+        }else{
```
Comment (Jakobzs) [inline lib/screens/settings_screens/settings_screen.dart:68]: Is the indentation correct here? In some other place it is "}else {"

File: lib/widgets/settings_widgets/settings_section_arrow_button.dart [hunk 3]
```diff
@@ -6,23 +6,42 @@ import '../../style/custom_color.dart' as theme;
...
         Icons.arrow_forward,
         color: theme.GirafColors.black,
       ),
       onTap: () => callback(),
     );
   }
+
+  /// Builds the text with or without the optional trailing widget
+  Widget buildTitle(){
+    if(titleTrailing == null){
```
Comment (Jakobzs) [inline lib/widgets/settings_widgets/settings_section_arrow_button.dart:35]: Should there be a space or not between the "{"

File: lib/widgets/settings_widgets/settings_section_colorThemeButton.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import 'package:weekplanner/widgets/settings_widgets/settings_section_item.dart';
+import 'package:weekplanner/widgets/settings_widgets/settings_theme_display_box.dart';
+import '../../style/custom_color.dart' as theme;
+
+/// Button used for the color theme settings
+class SettingsColorThemeCheckMarkButton extends SettingsSectionItem {
+  /// Constructor
+  const SettingsColorThemeCheckMarkButton(
+      this._expected, this._current, this._text, this._callback);
+
```
Comment (Jakobzs) [inline lib/widgets/settings_widgets/settings_section_colorThemeButton.dart:12]: Missing comments here?

File: lib/widgets/settings_widgets/settings_section_colorThemeButton.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        children: <Widget>[
+          ThemeBox.fromHexValues(_expected[0].hexColor, _expected[1].hexColor),
+          Text(_text),
+        ],
+      ),
+      trailing: trailing,
+      onTap: () => _callback(),
+    );
+  }
+
```
Comment (Jakobzs) [inline lib/widgets/settings_widgets/settings_section_colorThemeButton.dart:39]: The code is pretty self explanatory in my opinion but a few comments would not hurt, just to outline what happens here.

**Review by joandrsn — APPROVED** [review]
Looks good
